workflow/controller/dag.go

```diff
@@ -498,10 +499,17 @@ func (woc *wfOperationCtx) buildLocalScopeFromTask(dagCtx *dagContext, task *wfv
 		}
 		prefix := fmt.Sprintf("tasks.%s", ancestor)
 		if ancestorNode.Type == wfv1.NodeTypeTaskGroup {
+			// This happens when there was `withItem/withParam` expansion.
+			// We add the aggregate outputs of our children to the scope as a JSON list
 			var ancestorNodes []wfv1.NodeStatus
+			retryNodeNamePattern, _ := regexp.Compile(`.*\([0-9]+\)$`)
```

Comment:
This should be a `const` at the top of the file, so it's only called on once on program start up. Also, we can use `MustCompile`.

Something like

```go
const retryNodeNamePatter = regexp.MustCompile(`.*\([0-9]+\)$`)
```

At the top of this file is probably what we want. You should be able to find other examples of this in the codebase.

Reply:
> Sorry, I haven't had time to do a thorough review of this yet. But in my experience we should be very careful in extracting and inferring node types/relationships from their names. There can be a lot of edge cases. #5672 is an example of me dealing with this. In particular, take a look at the test cases on that PR. It is possible for user-defined output to be present on a node name, creating all sorts of edge cases when using regex or string manipulation to do these inferences, etc.
> 
> Again, I haven't had a chance to do a deep dive into this PR and confirm that this PR is safe for all edge cases (and even if I did, I wouldn't even be so confident with my own assertion anyway). Can you give me your thoughts on whether this is the most robust way to do this? Again, haven't done a deep dive here so I may be missing something.

Thanks. This is not the robust way to do this. And I found another bug about `Aggregated Outputs`: https://github.com/argoproj/argo-workflows/issues/6684. Currently, the outputs parameters of node with expansion expression is aggregated during processing childNode or grandChildNode. So the dag/steps outputs parameters cannot resolve these `aggregated outputs`. And i will these two issues in a new PR.